components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksInterceptor.kt

```diff
@@ -100,6 +101,12 @@ class AppLinksInterceptor(
         val result = handleRedirect(redirect, uri)
 
         if (redirect.isRedirect()) {
+            if (redirect.appIntent?.`package` != null &&
```

Comment:
Could we handle this in `AppLinksUseCases.safeParseUri` instead and return null there. Then we basically have the same fix as in Fennec, plus we don't add more conditions here. This is already quite hard to follow :).

Reply:
Understood.  I'll move the implementation into `safeParseUri`.  Thanks